### PR TITLE
Style: Change Char to char and Boolean to bool

### PR DESCRIPTION
**Pull request by iSazonov**
## PR Summary

Change Char to char and Boolean to bool.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [ ] Not Applicable or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [x] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: src/System.Management.Automation/engine/interpreter/CallInstruction.Generated.cs [hunk 1]
```diff
@@ -121,7 +121,7 @@ internal partial class CallInstruction {
                 case TypeCode.Int32: return FastCreate<Int32>(target, pi);
                 case TypeCode.Int64: return FastCreate<Int64>(target, pi);
                 case TypeCode.Boolean: return FastCreate<Boolean>(target, pi);
-                case TypeCode.Char: return FastCreate<Char>(target, pi);
+                case TypeCode.Char: return FastCreate<char>(target, pi);
```
Comment (daxian-dbw) [inline src/System.Management.Automation/engine/interpreter/CallInstruction.Generated.cs]: I think this one shouldn't be changed. We are dealing with primitive types in general here and the official type name is more readable than alias, e.g. `UInt64` vs. `ulong`. Better use official type name for all types here.

File: src/System.Management.Automation/engine/interpreter/NumericConvertInstruction.cs [hunk 2]
```diff
@@ -57,7 +57,7 @@ private object Convert(object obj)
                     case TypeCode.Byte: return ConvertInt32((Byte)obj);
                     case TypeCode.SByte: return ConvertInt32((SByte)obj);
                     case TypeCode.Int16: return ConvertInt32((Int16)obj);
-                    case TypeCode.Char: return ConvertInt32((Char)obj);
+                    case TypeCode.Char: return ConvertInt32((char)obj);
```
Comment (daxian-dbw) [inline src/System.Management.Automation/engine/interpreter/NumericConvertInstruction.cs]: Same here, better use official type names for all types here.

File: src/System.Management.Automation/engine/interpreter/NumericConvertInstruction.cs [hunk 3]
```diff
@@ -78,7 +78,7 @@ private object ConvertInt32(int obj)
                         case TypeCode.Byte: return (Byte)obj;
                         case TypeCode.SByte: return (SByte)obj;
                         case TypeCode.Int16: return (Int16)obj;
-                        case TypeCode.Char: return (Char)obj;
+                        case TypeCode.Char: return (char)obj;
```
Comment (daxian-dbw) [inline src/System.Management.Automation/engine/interpreter/NumericConvertInstruction.cs]: Same here.

File: src/System.Management.Automation/engine/interpreter/NumericConvertInstruction.cs [hunk 4]
```diff
@@ -100,7 +100,7 @@ private object ConvertInt64(Int64 obj)
...
```
Comment (daxian-dbw) [inline src/System.Management.Automation/engine/interpreter/NumericConvertInstruction.cs]: Same here.

File: src/System.Management.Automation/engine/interpreter/NumericConvertInstruction.cs [hunk 5]
```diff
@@ -122,7 +122,7 @@ private object ConvertUInt64(UInt64 obj)
...
```
Comment (daxian-dbw) [inline src/System.Management.Automation/engine/interpreter/NumericConvertInstruction.cs]: Same here.

File: src/System.Management.Automation/engine/interpreter/NumericConvertInstruction.cs [hunk 6]
```diff
@@ -144,7 +144,7 @@ private object ConvertDouble(Double obj)
...
```
Comment (daxian-dbw) [inline src/System.Management.Automation/engine/interpreter/NumericConvertInstruction.cs]: Same here.

File: src/System.Management.Automation/engine/interpreter/NumericConvertInstruction.cs [hunk 7]
```diff
@@ -202,7 +202,7 @@ private object ConvertInt32(int obj)
...
```
Comment (daxian-dbw) [inline src/System.Management.Automation/engine/interpreter/NumericConvertInstruction.cs]: Same here.

File: src/System.Management.Automation/engine/interpreter/NumericConvertInstruction.cs [hunk 8]
```diff
@@ -224,7 +224,7 @@ private object ConvertInt64(Int64 obj)
...
```
Comment (daxian-dbw) [inline src/System.Management.Automation/engine/interpreter/NumericConvertInstruction.cs]: Same here.

File: src/System.Management.Automation/engine/interpreter/NumericConvertInstruction.cs [hunk 9]
```diff
@@ -246,7 +246,7 @@ private object ConvertUInt64(UInt64 obj)
...
```
Comment (daxian-dbw) [inline src/System.Management.Automation/engine/interpreter/NumericConvertInstruction.cs]: Same here.

File: src/System.Management.Automation/engine/interpreter/NumericConvertInstruction.cs [hunk 10]
```diff
@@ -268,7 +268,7 @@ private object ConvertDouble(Double obj)
...
```
Comment (daxian-dbw) [inline src/System.Management.Automation/engine/interpreter/NumericConvertInstruction.cs]: Same here.

File: src/System.Management.Automation/engine/interpreter/Utilities.cs [hunk 11]
```diff
@@ -248,7 +248,7 @@ internal static object GetPrimitiveDefaultValue(Type type)
                 case TypeCode.Boolean: return ScriptingRuntimeHelpers.False;
                 case TypeCode.SByte: return default(SByte);
                 case TypeCode.Byte: return default(Byte);
-                case TypeCode.Char: return default(Char);
+                case TypeCode.Char: return default(char);
```
Comment (daxian-dbw) [inline src/System.Management.Automation/engine/interpreter/Utilities.cs]: Same here.

**Review by daxian-dbw — CHANGES_REQUESTED** [review]

File: src/System.Management.Automation/engine/interpreter/CallInstruction.Generated.cs (hunk 1 above)
Comment (iSazonov) [reply src/System.Management.Automation/engine/interpreter/CallInstruction.Generated.cs]:
```suggestion
                case TypeCode.Char: return FastCreate<Char>(target, pi);
```

Comment (iSazonov) [reply src/System.Management.Automation/engine/interpreter/CallInstruction.Generated.cs]:
```suggestion
                case TypeCode.Char: return FastCreate<Char>(target, pi);
```

File: src/System.Management.Automation/engine/interpreter/NumericConvertInstruction.cs (hunk 2 above)
Comment (iSazonov) [reply src/System.Management.Automation/engine/interpreter/NumericConvertInstruction.cs]:
```suggestion
                    case TypeCode.Char: return ConvertInt32((Char)obj);
```

File: src/System.Management.Automation/engine/interpreter/NumericConvertInstruction.cs (hunk 3 above)
Comment (iSazonov) [reply src/System.Management.Automation/engine/interpreter/NumericConvertInstruction.cs]:
```suggestion
                        case TypeCode.Char: return (Char)obj;
```

File: src/System.Management.Automation/engine/interpreter/NumericConvertInstruction.cs (hunk 4 above)
Comment (iSazonov) [reply src/System.Management.Automation/engine/interpreter/NumericConvertInstruction.cs]:
```suggestion
                        case TypeCode.Char: return (Char)obj;
```

File: src/System.Management.Automation/engine/interpreter/NumericConvertInstruction.cs (hunk 5 above)
Comment (iSazonov) [reply src/System.Management.Automation/engine/interpreter/NumericConvertInstruction.cs]:
```suggestion
                        case TypeCode.Char: return (Char)obj;
```

File: src/System.Management.Automation/engine/interpreter/NumericConvertInstruction.cs (hunk 6 above)
Comment (iSazonov) [reply src/System.Management.Automation/engine/interpreter/NumericConvertInstruction.cs]:
```suggestion
                        case TypeCode.Char: return (Char)obj;
```

File: src/System.Management.Automation/engine/interpreter/NumericConvertInstruction.cs (hunk 7 above)
Comment (iSazonov) [reply src/System.Management.Automation/engine/interpreter/NumericConvertInstruction.cs]:
```suggestion
                        case TypeCode.Char: return (Char)obj;
```

File: src/System.Management.Automation/engine/interpreter/NumericConvertInstruction.cs (hunk 8 above)
Comment (iSazonov) [reply src/System.Management.Automation/engine/interpreter/NumericConvertInstruction.cs]:
```suggestion
                        case TypeCode.Char: return (Char)obj;
```

File: src/System.Management.Automation/engine/interpreter/NumericConvertInstruction.cs (hunk 9 above)
Comment (iSazonov) [reply src/System.Management.Automation/engine/interpreter/NumericConvertInstruction.cs]:
```suggestion
                        case TypeCode.Char: return (Char)obj;
```

File: src/System.Management.Automation/engine/interpreter/NumericConvertInstruction.cs (hunk 10 above)
Comment (iSazonov) [reply src/System.Management.Automation/engine/interpreter/NumericConvertInstruction.cs]:
```suggestion
                        case TypeCode.Char: return (Char)obj;
```

**Comment by iSazonov** [comment]
@daxian-dbw Your comments is addressed and the merge conflict is resolved.

**Review by daxian-dbw — APPROVED** [review]
LGTM

**Comment by daxian-dbw** [comment]
There is another conflict after merging your 'String-to-string' PR. Can you please resolve this one too?

**Comment by iSazonov** [comment]
Done.

**Review by daxian-dbw — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**